packages/__docs__/src/Preview/index.tsx

```diff
@@ -154,5 +142,35 @@ class Preview extends Component<PreviewProps, PreviewState> {
   }
 }
 
+class PreviewErrorBoundary extends Component<
+  React.PropsWithChildren,
+  { error: string | null }
+> {
+  constructor(props: any) {
+    super(props)
+    this.state = { error: null }
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    return { error: error.toString() }
+  }
+
+  render() {
+    const children = React.Children.map(this.props.children, (child) => {
+      if (React.isValidElement(child) && this.state.error) {
+        return React.cloneElement(child, {
+          ...child.props,
+          // have to set this to empty string to prevent infinte render cycle
+          code: '',
+          error: this.state.error
+        })
+      }
+      return child
+    })
+
+    return children
+  }
+}
+
```

Comment:
This component had to be added since the code coming from `compileAndRenderExample` could contain invalid code and that invalid code would cause an error during rendering inside the `Preview` component. To prevent that from happening, this `PreviewErrorBoundary` will 'catch' the error, then passes down to the `Preview` and then `Preview` can handle the error as usual.